src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextJsonReader.cs

```diff
@@ -2,16 +2,42 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
+using System.Buffers;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.IO;
 using System.Linq;
+using System.Text.Json;
 
 namespace Microsoft.Extensions.DependencyModel
 {
-    public partial class DependencyContextJsonReader : IDependencyContextReader
+    public class DependencyContextJsonReader : IDependencyContextReader
     {
+        private const int UnseekableStreamInitialRentSize = 4096;
+        private static ReadOnlySpan<byte> Utf8Bom => new byte[] { 0xEF, 0xBB, 0xBF };
+
         private readonly IDictionary<string, string> _stringPool = new Dictionary<string, string>();
 
+        public DependencyContext Read(Stream stream)
+        {
+            if (stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+
+            ArraySegment<byte> buffer = ReadToEnd(stream);
```

Comment:
@steveharter - do you know why `Utf8JsonReader` doesn't support a `Stream` in its ctor? `Utf8JsonWriter` does.

Reply:
Probably because a best practice it to access Stream async (instead of draining upfront), and it's not possible to have async methods with a `ref struct` like `Utf8JsonReader`. So to do this "correctly" with Stream+async you have to write a bit of code today that uses `JsonReaderState`. Also see https://github.com/dotnet/runtime/issues/30328

The deserializer support Stream in an async way however, if that's a better option.

cc @ahsonkhan

Reply:
Right :)

A stream (or pipelines) based reader can be built on top of the span-based ref struct reader.